Assets/LeapMotionModules/Interaction/InteractionHand.cs

```diff
@@ -0,0 +1,800 @@
+    //internal void ContactBoneTriggerEnter(ContactBone contactBone, InteractionBehaviourBase interactionObjr) {
+    //}
+
+    //internal void ContactBoneTriggerExit(ContactBone contactBone, InteractionBehaviourBase interactionObj) {
+    //}
+
+    private void FixedUpdateContactCallbacks() {
+      foreach (var interactionObj in _contactBehavioursLastFrame) {
+        if (!_contactBehaviours.ContainsKey(interactionObj)
+          || !_brushBoneParent.gameObject.activeSelf) {
+          interactionObj.ContactEnd(_hand);
+          _contactBehaviourRemovalCache.Add(interactionObj);
+        }
+      }
+      foreach (var interactionObj in _contactBehaviourRemovalCache) {
+        _contactBehavioursLastFrame.Remove(interactionObj);
+      }
+      _contactBehaviourRemovalCache.Clear();
+      if (_brushBoneParent.gameObject.activeSelf) {
+        foreach (var intObjCountPair in _contactBehaviours) {
+          var interactionObj = intObjCountPair.Key;
+          if (_contactBehavioursLastFrame.Contains(interactionObj)) {
+            interactionObj.ContactStay(_hand);
+          }
+          else {
+            interactionObj.ContactBegin(_hand);
+            _contactBehavioursLastFrame.Add(interactionObj);
+          }
+        }
+      }
+    }
+
+    #endregion
+
+    #endregion
+
+    #region Grasping
+
+    private HeuristicGrabClassifier __grabClassifier;
+    private HeuristicGrabClassifier _grabClassifier {
+      get {
+        if (__grabClassifier == null) __grabClassifier = new HeuristicGrabClassifier(this);
+        return __grabClassifier;
+      }
+    }
+
+    private InteractionBehaviourBase _graspedObject;
+
+    private void FixedUpdateGrasping() {
+      _grabClassifier.FixedUpdateHeuristicClassifier(_hand);
+
+      if (_graspedObject != null) {
+        if (!_graspedObject.allowGrasping) {
+          ReleaseGrasp();
+        }
+        _graspedObject.GraspHold(_hand);
+      }
+    }
+
+    public void Grasp(InteractionBehaviourBase interactionObj) {
+      if (!interactionObj.allowGrasping) return;
+
+      if (_graspedObject != null) {
+        Debug.LogError("Grasp was called, but the hand was already grasping an object.");
+        ReleaseGrasp();
+      }
+
+      interactionObj.GraspBegin(_hand);
+      _graspedObject = interactionObj;
+    }
+
+    public void ReleaseGrasp() {
+      if (_graspedObject == null) {
+        Debug.LogWarning("ReleaseGrasp() was called, but there is no held object to release.");
+        return; // Nothing to release.
+      }
+
+      // Enable Soft Contact if contact in general is enabled to prevent objects
+      // popping right out of the hand.
+      if (_contactInitialized && interactionManager.enableContact) {
+        EnableSoftContact();
+      }
+
+      _grabClassifier.NotifyGraspReleased(_graspedObject);
+      _graspedObject.GraspEnd(_hand);
+      _graspedObject = null;
+    }
+
+    public bool ReleaseObject(InteractionBehaviourBase interactionObj) {
```

Comment:
Open for discussion, but I would argue that any method that can fail should be prefixed with a Try to make it obvious it can fail.